src/types/web-compat.ts

```diff
@@ -0,0 +1,51 @@
+// eslint-disable
+/**
+ * @module Webcompat Messages Schema
+ * @description
+ * These types are auto-generated from schema files.
+ * scripts/build-types.mjs is responsible for type generation.
+ * See the privacy-configuration repo for the schema files:
+ * https://github.com/duckduckgo/privacy-configuration
+ * **DO NOT** edit this file directly as your changes will be lost.
+ */
+
+/**
+ * Requests, Notifications and Subscriptions from the Web Compat feature
+ */
+export interface WebCompatMessages {
+  requests: WebShareMessage;
+}
+/**
+ * todo: add description for `webShare` message
+ */
+export interface WebShareMessage {
+  method: "webShare";
+  params: WebShareParams;
+  /**
+   * todo: add return type here
+   */
+  result: {
+    [k: string]: unknown;
+  };
+}
+export interface WebShareParams {
+  /**
+   * todo: add description for 'title' field
+   */
+  title?: string;
+  /**
+   * todo: add description for 'url' field
+   */
+  url?: string;
+  /**
+   * todo: add description for 'text' field
+   */
+  text?: string;
+}
+
+declare module "../features/web-compat.js" {
+  export interface WebCompat {
+    request: MessagingBase<WebCompatMessages>['request']
+  }
+}
+
```

Comment:
this is the 'magic' 🪄 - here we are appending the `request` method to `WebCompat` in a type-safe way. Then, because `request` in the base class is typed in a loose manner, this one 'wins' and the type safety is achieved in the feature-file.

Reply:
(same will apply for notifications and subscriptions too)

Reply:
I see, so if we had messages that aren't feature specific we might need to do some further work to support that?

Reply:
I'm looking now before we merge if there's any tweaks to make it easier to support none-features